### ISSUE TITLE
Should the standard switch to flat rather than nested data?

### ISSUE THREAD

**Issue by hughstimson** [PlaceSpeak/LUCD#12]
The [BLDS](https://github.com/open-data-standards/permitdata.org/wiki) standard (and as far as I can tell most standards), seem to built with the assumption that the default formatting of the data will be as a simple table with a un-changing number of columns. E.g.

```
AppliedDate, IssuedDate, CompletedDate
2015-01-01,  2015-02-01, 2015-03-01
```

As it is written, LUCD allows for repeating instances of certain types of data, e.g. status dates, addresses. E.g.

```
statuses [
    [ name: "applied", date: "2015-01-01"],
    [ name: "revised", date: "2015-01-15"],
    [ name: "issued", date: "2015-02-01"],
    [ name: "completed", date: 2015-03-01"]
]
```
(The above is in JSON formatting but it could likewise be presented in XML, as could the first example.)

The benefits of repeatable data is that more information can be included more flexibly, and that information is sometimes going to be important. There's often more than one affected address! How many address columns would we need to include? Address1, Address2, Address3,...?

The benefit of non-repeating data is that display is more straightforward, and the data becomes more predictable in character and as such easier to parse.

Maybe it's simply the case that common portals such as Socrata and CKAN aren't even set up to store and display more complex data formats? That would de facto resolve the question.

**Comment by hughstimson** [comment]
Having looked further into the way BLDS handles this, it looks like there's a compromise option they're using.

BLDS requires a core "Permits" table, then optionally suggests some additional tables for information where there might be more than one entry per individual permit.

E.g.

Core Permits Dataset
PermitNum|Description|...
---------|-----------|---
1000     |New deck   |...
1001     |Addition   |...

Optional Status Change Dataset
PermitNum|Status  |Date    |...
---------|--------|--------|---
1000     |Received|1-1-2016|...
1000     |Approved|1-2-2016|...
1001     |Received|1-2-2016|...
1002     |Denied  |1-3-2016|...

to make a sort of jury rigged relational database. Seems a little hacky, but presumably it works and is portable to various publishing methods.

This represents a significant change to the standard. I'll include it in the 0.9 release.

**Comment by hughstimson** [comment]
Closed by pull request #14.
